lib/puppet/provider/templates/invoke_dsc_resource.ps1.erb

```diff
@@ -21,7 +21,9 @@ $invokeParams = @{
 <% dsc_parameters.each do |p| -%>
     <%- name = p.name.to_s.gsub(/^dsc_/,'')
     value = format_dsc_value(p.value)
-    value = '\'present\'' if name == 'ensure' && dsc_invoke_method == 'test'
+    if name == 'ensure' && dsc_invoke_method == 'test'
+      value = "'#{resource.provider.ensure_value}'"
+    end
```

Comment:
Now that we've proven this out, let's flip it back like it was before with the if statement tacked on to setting the value.